abc-bank/Transactions/Transfer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+
+using abc_bank.Accounts;
+using abc_bank.Exceptions;
+using abc_bank.Utilities;
+
+namespace abc_bank.Transactions {
+  public class Transfer : ITransfer {
+
+    #region Properties
+
+    public DateTime Date { get; }
+
+    public double Amount { get; }
+
+    public int OriginAccountId { get; }
+
+    public int DestinationAccountId { get; }
+
+    #endregion
+
+    #region CTOR
+
+    public Transfer(double transferAmount, DateTime transactionDate, IAccount originAccount, IAccount destinationAccount) {
+
+      #region Validation
+
+      if (originAccount.CustomerId != destinationAccount.CustomerId) {
+        throw new InvalidTransactionRequestException();
+      }
+
+      Validators.FundsAvailableForWithdraw(originAccount.CurrentBalance, transferAmount);
+
+      #endregion
+
+      Amount = transferAmount;
+      Date = transactionDate;
+      OriginAccountId = originAccount.Id;
+      DestinationAccountId = destinationAccount.Id;
+
+      // process the transfer as a normal Withdraw/Deposit - with some addtional information
```

Comment:
Real world, I'd also have included a try-fail-rollback system.  As-is now, if the deposit fails the money is 'lost'